cdc/metadata.cc

```diff
@@ -86,33 +93,31 @@ cdc::stream_id cdc::metadata::get_stream(api::timestamp_type ts, dht::token tok)
         // Nothing protects us from that until we start using transactions for generation switching.
     }
 
-    auto it = gen_used_at(now);
+    auto it = gen_used_at(ts);
     if (it == _gens.end()) {
         throw std::runtime_error(format(
-                "cdc::metadata::get_stream: could not find any CDC stream (current time: {})."
-                " Are we in the middle of a cluster upgrade?", format_timestamp(now)));
+                "cdc::metadata::get_stream: could not find any CDC stream ({})."
+                " Are we in the middle of a cluster upgrade?", format_timestamp(ts)));
     }
 
-    // Garbage-collect generations that will no longer be used.
-    it = _gens.erase(_gens.begin(), it);
+    auto current_gen = it == gen_used_at(now);
```

Comment:
I initially got confused thinking that `current_gen` is an iterator type. Maybe rename it to `is_current_gen` and even better, replace the `auto` with `bool`?

Reply:
This comment became outdated in v2.